src/store/actions.js

```diff
@@ -64,12 +65,27 @@ export default {
   updateLatestTransactions({ commit }, payload) {
     commit(types.UPDATE_LATEST_TRANSACTIONS, payload);
   },
-  setAccountName({ commit }, payload) {
-    commit(types.SET_ACCOUNT_AENS, {
-      account: 0,
-      aename: payload.name,
-      pending: false,
-    });
+  async setAccountName({ commit, state }, { account = 0, aename = null, pending = false }) {
+    commit(types.SET_ACCOUNT_AENS, { account, aename, pending });
+
+    if (aename) {
+      try {
+        const response = await Backend.sendProfileData({
+          author: state.account.publicKey,
+          preferredChainName: aename,
+        });
+        const signedChallenge = Buffer.from(
+          await state.sdk.signMessage(response.challenge),
+        ).toString('hex');
+        const respondChallenge = {
+          challenge: response.challenge,
+          signature: signedChallenge,
+        };
+        await Backend.sendProfileData(respondChallenge);
+      } catch (e) {
+        console.warn(`Cannot set preferred name ${e}`);
```

Comment:
Maybe add error popup here?

Reply:
Isn't it a rare case? Maybe it would be better to make a global error popup? (to show it after any uncatched exception)

Reply:
This warning can be used only by developers, I don't think that it is more useful than the usual exception stack trace. So, I proposing to drop try/catch at all.